src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -511,10 +532,74 @@ void nl_Chip_DeviceController_SetLogFilter(uint8_t category)
 #endif
 }
 
+CHIP_ERROR nl_Chip_DeviceController_Connect(chip::DeviceController::ChipDeviceController * devCtrl, BLE_CONNECTION_OBJECT connObj,
+                                            uint32_t setupPINCode, OnConnectFunct onConnect, OnMessageFunct onMessage,
+                                            OnErrorFunct onError)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::RendezvousParameters params =
+        chip::RendezvousParameters().SetSetupPINCode(setupPINCode).SetConnectionObject(connObj).SetBleLayer(&sBle);
+    ;
+    err = devCtrl->ConnectDevice(kRemoteDeviceId, params, (void *) devCtrl, onConnect, onMessage, onError);
+    SuccessOrExit(err);
+exit:
+    return err;
+}
+
+CHIP_ERROR
+nl_Chip_ScriptDevicePairingDelegate_NewPairingDelegate(chip::DeviceController::ScriptDevicePairingDelegate ** pairingDelegate)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(pairingDelegate != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    *pairingDelegate = new chip::DeviceController::ScriptDevicePairingDelegate();
+
+exit:
+    if (err != CHIP_NO_ERROR && pairingDelegate != nullptr && *pairingDelegate != nullptr)
```

Comment:
if chip is no error, pairing delegate cannot be nullptr.
However we do not check if the passed in *pairingDelegate is nullptr.
In this case I think the exit-styl code makes it harder to read. I would find this clearer to read:

```
if (pairingDelegate == nullptr) {
  return CHIP_ERROR_INVALID_ARGUMENT;
}

*pairingDelegate = chip::Platform::New<chip::DeviceController::ScriptDevicePairingDelegate>();
return CHIP_NO_ERROR;
```

thoughts? Note that I also used the platform new ... unsure if it applies here.